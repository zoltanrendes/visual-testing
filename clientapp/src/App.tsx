import React, { useState } from "react";
import {
  Button,
  Row,
  Col,
  Navbar,
  Nav,
  Container,
  NavbarToggler,
  NavbarBrand,
  Collapse,
  NavItem,
  NavLink,
  Form,
  FormGroup,
  Input,
  Jumbotron,
  Card,
} from "reactstrap";

import logo605413 from "./assets/wet_phone_605413.jpg";
import logo605537 from "./assets/mobile_on_desk_605537.jpg";
import logo270068 from "./assets/phone_ancient_old_270068.jpg";
import logo596417 from "./assets/black_iphone_on_light_wood_596417.jpg";
import logo604599 from "./assets/phone_on_table_604599.jpg";
import logo183280 from "./assets/phone_183280.jpg";

import "./styles/App.scss";

function App() {
  const [isOpen, setIsOpen] = useState(false);

  const toggle = () => setIsOpen(!isOpen);

  return (
    <Container className="py-3 text-center">
      <Jumbotron>
        <Container className="text-center">
          <h1>Online Store</h1>
          <p>Mission, Vission & Values</p>
        </Container>
      </Jumbotron>

      <Navbar color="dark" dark expand="md">
        <NavbarBrand href="/">Shop</NavbarBrand>
        <NavbarToggler onClick={toggle} />
        <Collapse isOpen={isOpen} navbar>
          <Nav className="mr-auto" navbar>
            <NavItem>
              <NavLink href="/components/">Products</NavLink>
            </NavItem>
            <NavItem>
              <NavLink href="/components/">Deals</NavLink>
            </NavItem>
            <NavItem>
              <NavLink href="/components/">Stores</NavLink>
            </NavItem>
            <NavItem>
              <NavLink href="/components/">Contact</NavLink>
            </NavItem>
          </Nav>
        </Collapse>
      </Navbar>

      <Container className="my-2">
        <Row>
          <Col sm={4}>
            <Card>
              <div className="panel panel-primary">
                <div className="panel-heading">BLACK FRIDAY DEAL</div>
                <div className="panel-body">
                  <img src={logo605413} className="img-responsive" style={{ width: "100%" }} alt="somethingCool" />
                </div>
                <div className="panel-footer">Buy 50 mobiles and get a gift card</div>
              </div>
            </Card>
          </Col>
          <Col sm={4}>
            <Card>
              <div className="panel panel-primary">
                <div className="panel-heading">BLACK FRIDAY DEAL</div>
                <div className="panel-body">
                  <img src={logo605537} className="img-responsive" style={{ width: "100%" }} alt="somethingCool" />
                </div>
                <div className="panel-footer">Buy 50 mobiles and get a gift card</div>
              </div>
            </Card>
          </Col>
          <Col sm={4}>
            <Card>
              <div className="panel panel-primary">
                <div className="panel-heading">BLACK FRIDAY DEAL</div>
                <div className="panel-body">
                  <img src={logo270068} className="img-responsive" style={{ width: "100%" }} alt="somethingCool" />
                </div>
                <div className="panel-footer">Buy 50 mobiles and get a gift card</div>
              </div>
            </Card>
          </Col>
        </Row>
      </Container>

      <Container>
        <Row>
          <Col sm={4}>
            <div className="panel panel-primary">
              <div className="panel-heading">BLACK FRIDAY DEAL</div>
              <div className="panel-body">
                <img src={logo596417} className="img-responsive" style={{ width: "100%" }} alt="somethingCool" />
              </div>
              <div className="panel-footer">Buy 50 mobiles and get a gift card</div>
            </div>
          </Col>
          <Col sm={4}>
            <div className="panel panel-primary">
              <div className="panel-heading">BLACK FRIDAY DEAL</div>
              <div className="panel-body">
                <img src={logo604599} className="img-responsive" style={{ width: "100%" }} alt="somethingCool" />
              </div>
              <div className="panel-footer">Buy 50 mobiles and get a gift card</div>
            </div>
          </Col>
          <Col sm={4}>
            <div className="panel panel-primary">
              <div className="panel-heading">BLACK FRIDAY DEAL</div>
              <div className="panel-body">
                <img src={logo183280} className="img-responsive" style={{ width: "100%" }} alt="somethingCool" />
              </div>
              <div className="panel-footer">Buy 50 mobiles and get a gift card</div>
            </div>
          </Col>
        </Row>
      </Container>
      <Container className="text-center my-3">
        <Row>
          <Col sm="12">
            <footer>
              <p>Online Store Copyright</p>
              <Row className="text-center">
                <Col md={{ size: 12, offset: 3 }}>
                  <Form inline>
                    <FormGroup>
                      Get deals:
                      <Input type="email" className="form-control mx-2" size={50} placeholder="Email Address" />
                      <Button type="button" className="btn btn-success">
                        Sign Up
                      </Button>
                    </FormGroup>
                  </Form>
                </Col>
              </Row>
            </footer>
          </Col>
        </Row>
      </Container>
    </Container>
  );
}

export default App;
