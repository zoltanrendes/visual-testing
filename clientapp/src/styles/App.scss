// please, uncomment to see failing
// #root {
//   transform: rotate(180deg);
// }
