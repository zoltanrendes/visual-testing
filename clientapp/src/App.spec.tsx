import React from "react";
import { render, screen } from "@testing-library/react";
import App from "./App";

describe("App.tsx", () => {
  it("should render all elements", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();

    expect(screen.getByRole("navigation").textContent?.includes("Shop")).toBeTruthy();
    expect(screen.getByRole("navigation").textContent?.includes("Products")).toBeTruthy();
    expect(screen.getByRole("navigation").textContent?.includes("Deals")).toBeTruthy();
    expect(screen.getByRole("navigation").textContent?.includes("Stores")).toBeTruthy();
    expect(screen.getByRole("navigation").textContent?.includes("Contact")).toBeTruthy();

    expect(screen.getAllByRole("navigation")).toHaveLength(1);
    expect(screen.getAllByRole("img")).toHaveLength(6);
    expect(screen.getAllByRole("textbox")).toHaveLength(1);
    expect(screen.getAllByRole("button")).toHaveLength(2);

    const buttons = screen.getAllByRole("button").values();

    expect((buttons.next().value as HTMLButtonElement).className).toBe("navbar-toggler");
    expect((buttons.next().value as HTMLButtonElement).textContent).toBe("Sign Up");

    expect(screen.getByText(/Online Store Copyright/i)).toBeTruthy();
    expect(screen.getAllByPlaceholderText("Email Address")).toHaveLength(1);
  });
});
