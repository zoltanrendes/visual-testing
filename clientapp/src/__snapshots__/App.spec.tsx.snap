// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.tsx should render all elements 1`] = `
<div>
  <div
    class="py-3 text-center container"
  >
    <div
      class="jumbotron"
    >
      <div
        class="text-center container"
      >
        <h1>
          Online Store
        </h1>
        <p>
          Mission, Vission & Values
        </p>
      </div>
    </div>
    <nav
      class="navbar navbar-expand-md navbar-dark bg-dark"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        Shop
      </a>
      <button
        aria-label="Toggle navigation"
        class="navbar-toggler"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
      >
        <ul
          class="mr-auto navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/components/"
            >
              Products
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/components/"
            >
              Deals
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/components/"
            >
              Stores
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/components/"
            >
              Contact
            </a>
          </li>
        </ul>
      </div>
    </nav>
    <div
      class="my-2 container"
    >
      <div
        class="row"
      >
        <div
          class="col-sm-4"
        >
          <div
            class="card"
          >
            <div
              class="panel panel-primary"
            >
              <div
                class="panel-heading"
              >
                BLACK FRIDAY DEAL
              </div>
              <div
                class="panel-body"
              >
                <img
                  alt="somethingCool"
                  class="img-responsive"
                  src="wet_phone_605413.jpg"
                  style="width: 100%;"
                />
              </div>
              <div
                class="panel-footer"
              >
                Buy 50 mobiles and get a gift card
              </div>
            </div>
          </div>
        </div>
        <div
          class="col-sm-4"
        >
          <div
            class="card"
          >
            <div
              class="panel panel-primary"
            >
              <div
                class="panel-heading"
              >
                BLACK FRIDAY DEAL
              </div>
              <div
                class="panel-body"
              >
                <img
                  alt="somethingCool"
                  class="img-responsive"
                  src="mobile_on_desk_605537.jpg"
                  style="width: 100%;"
                />
              </div>
              <div
                class="panel-footer"
              >
                Buy 50 mobiles and get a gift card
              </div>
            </div>
          </div>
        </div>
        <div
          class="col-sm-4"
        >
          <div
            class="card"
          >
            <div
              class="panel panel-primary"
            >
              <div
                class="panel-heading"
              >
                BLACK FRIDAY DEAL
              </div>
              <div
                class="panel-body"
              >
                <img
                  alt="somethingCool"
                  class="img-responsive"
                  src="phone_ancient_old_270068.jpg"
                  style="width: 100%;"
                />
              </div>
              <div
                class="panel-footer"
              >
                Buy 50 mobiles and get a gift card
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-sm-4"
        >
          <div
            class="panel panel-primary"
          >
            <div
              class="panel-heading"
            >
              BLACK FRIDAY DEAL
            </div>
            <div
              class="panel-body"
            >
              <img
                alt="somethingCool"
                class="img-responsive"
                src="black_iphone_on_light_wood_596417.jpg"
                style="width: 100%;"
              />
            </div>
            <div
              class="panel-footer"
            >
              Buy 50 mobiles and get a gift card
            </div>
          </div>
        </div>
        <div
          class="col-sm-4"
        >
          <div
            class="panel panel-primary"
          >
            <div
              class="panel-heading"
            >
              BLACK FRIDAY DEAL
            </div>
            <div
              class="panel-body"
            >
              <img
                alt="somethingCool"
                class="img-responsive"
                src="phone_on_table_604599.jpg"
                style="width: 100%;"
              />
            </div>
            <div
              class="panel-footer"
            >
              Buy 50 mobiles and get a gift card
            </div>
          </div>
        </div>
        <div
          class="col-sm-4"
        >
          <div
            class="panel panel-primary"
          >
            <div
              class="panel-heading"
            >
              BLACK FRIDAY DEAL
            </div>
            <div
              class="panel-body"
            >
              <img
                alt="somethingCool"
                class="img-responsive"
                src="phone_183280.jpg"
                style="width: 100%;"
              />
            </div>
            <div
              class="panel-footer"
            >
              Buy 50 mobiles and get a gift card
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="text-center my-3 container"
    >
      <div
        class="row"
      >
        <div
          class="col-sm-12"
        >
          <footer>
            <p>
              Online Store Copyright
            </p>
            <div
              class="text-center row"
            >
              <div
                class="col-md-12 offset-md-3"
              >
                <form
                  class="form-inline"
                >
                  <div
                    class="form-group"
                  >
                    Get deals:
                    <input
                      class="form-control mx-2 form-control"
                      placeholder="Email Address"
                      size="50"
                      type="email"
                    />
                    <button
                      class="btn btn-success btn btn-secondary"
                      type="button"
                    >
                      Sign Up
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </footer>
        </div>
      </div>
    </div>
  </div>
</div>
`;
